geth/node/syncpoll.go

```diff
@@ -0,0 +1,76 @@
+package node
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum/go-ethereum/eth/downloader"
+	"github.com/ethereum/go-ethereum/les"
+)
+
+// errors
+var (
+	ErrNodeSyncFailedToStart = errors.New("node synchronization failed to start")
+	ErrNodeSyncTakesTooLong  = errors.New("node synchronization is taking too long")
+)
+
+// delays
+var (
+	// delayCycleForSyncStart sets the timeout for checking state of synchronization starting.
+	delayCycleForSyncStart = 100 * time.Millisecond
+)
+
+// SyncPoll provides a structure that allows us to check the status of
+// ethereum node synchronization.
+type SyncPoll struct {
+	downloader *downloader.Downloader
+}
+
+// NewSyncPoll returns a new instance of SyncPoll.
+func NewSyncPoll(leth *les.LightEthereum) *SyncPoll {
+	return &SyncPoll{
+		downloader: leth.Downloader(),
+	}
+}
+
+// Poll returns a channel which allows the user to listen for a done signal
+// as to when the node has finished syncing or stop due to an error.
+func (n *SyncPoll) Poll(ctx context.Context) error {
+	if err := n.pollSyncStart(ctx); err != nil {
+		return err
+	}
+
+	if err := n.pollSyncCompleted(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *SyncPoll) pollSyncStart(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ErrNodeSyncFailedToStart
+		case <-time.After(delayCycleForSyncStart):
```

Comment:
I think this timeout is not even worth having a separate name as one can't figure out what it does looking at their name. At the same time, you need to jump to that variable to see what it contains being on this line. I think, having a concrete number here would be meaningful. What do you think?

The same about `pollSyncCompleted`.